testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -4537,6 +4537,27 @@ public void testUpdateWithPredicates()
         }
     }
 
+    @Test
+    public void testUpdateWithPredicatesForVarchars()
+    {
+        if (!hasBehavior(SUPPORTS_UPDATE)) {
+            // Note this change is a no-op, if actually run
+            assertQueryFails("UPDATE nation SET nationkey = nationkey + regionkey WHERE regionkey < 1", MODIFYING_ROWS_MESSAGE);
+            return;
+        }
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_update_with_predicates_varchar", "(a INT, b INT, c INT, d VARCHAR)")) {
+            String tableName = table.getName();
+            assertUpdate("INSERT INTO " + tableName + " VALUES (1, 2, 3, 'var1'), (11, 12, 13, 'var2'), (21, 22, 23, 'var3')", 3);
+
+            assertUpdate("UPDATE " + tableName + " SET b = 44 WHERE d = 'var3'", 1);
```

Comment:
Add test about the case sensitive inequality predicates,like `BaseJdbcConnectorTest.testDeleteWithVarcharInequalityPredicate` and `BaseJdbcConnectorTest.testDeleteWithVarcharGreaterAndLowerPredicate`

Reply:
We are not handling such cases in this PR, it's our next step.
in case of inequality or greater lower predicates
execution plan will be different and we don't execute this code at all.

Reply:
will add this to description

Reply:
tests added